src/init.cpp

```diff
@@ -739,16 +739,7 @@ bool AppInit2(ThreadHandlerPtr threads)
 #endif
 #ifdef WIN32
     // Enable Data Execution Prevention (DEP)
-    // Minimum supported OS versions: WinXP SP3, WinVista >= SP1, Win Server 2008
-    // A failure is non-critical and needs no further attention!
-#ifndef PROCESS_DEP_ENABLE
-// We define this here, because GCCs winbase.h limits this to _WIN32_WINNT >= 0x0601 (Windows 7),
-// which is not correct. Can be removed, when GCCs winbase.h is fixed!
-#define PROCESS_DEP_ENABLE 0x00000001
-#endif
-    typedef BOOL (WINAPI *PSETPROCDEPPOL)(DWORD);
-    PSETPROCDEPPOL setProcDEPPol = (PSETPROCDEPPOL)GetProcAddress(GetModuleHandleA("Kernel32.dll"), "SetProcessDEPPolicy");
-    if (setProcDEPPol != nullptr) setProcDEPPol(PROCESS_DEP_ENABLE);
+    SetProcessDEPPolicy(PROCESS_DEP_ENABLE);
```

Comment:
This doesn't enable DEP on 64-bit machines, better to use `/NXCOMPAT`, `--nxcompat` linker options instead.

Reply:
Is your suggestion done in Bitcoin upstream?

Reply:
And... the older code removed would have had the same problem, I think. So we have not had DEP for the 64 bit version, or maybe we have, since DEP is turned on by default on 64 bit Windows I think.

Reply:
https://github.com/bitcoin/bitcoin/commit/3d5d7aad269c7afe7e36677d3e76c6579e1b8aba
https://github.com/bitcoin/bitcoin/blob/9c3751a0c940228c46461c4191f1a57153cb9037/configure.ac#L911
and according to https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getsystemdeppolicy OptIn is the default.

Reply:
That AX_CHECK_LINK_FLAG([[-Wl,--nxcompat]], [HARDENED_LDFLAGS="$HARDENED_LDFLAGS -Wl,--nxcompat"],, [[$LDFLAG_WERROR]]) is already in our configure so it looks like we should just remove the call to SetProcessDEPPolicy. I am adding the terminate on heap corruption call. That seems like a good idea.